source/geometries/NextDemoSiPMBoard.cc

```diff
@@ -167,31 +207,29 @@ void NextDemoSiPMBoard::Construct()
 
 
   /// Mask Holes
+  G4VSolid* hole_solid = nullptr;
   G4String hole_name = "BOARD_MASK_HOLE";
-
-  G4Tubs* hole_solid = new G4Tubs(hole_name, 0., hole_diam_/2.,
-                                  mask_thickn_/2., 0, 360.*deg);
+  if (hole_type_ == "rounded"){
+    hole_solid = new G4Tubs(hole_name, 0., hole_diam_/2., mask_thickn_/2., 0, 360.*deg);}
+  else if (hole_type_ == "rectangular"){
+    hole_solid = new G4Box(hole_name, hole_x_/2., hole_y_/2., mask_thickn_/2.);}
 
   G4LogicalVolume* hole_logic =
     new G4LogicalVolume(hole_solid, mother_gas, hole_name);
 
-
-  /// SiPMs construction
-  sipm_->SetSensorDepth(3);
-  sipm_->SetMotherDepth(5);
-  sipm_->SetNamingOrder(1000);
-  sipm_->Construct();
-
   // Generate SiPM positions
   GenerateSiPMPositions();
 
   // SiPM placement inside the hole
   G4RotationMatrix* sipm_rot = new G4RotationMatrix();
-  sipm_rot->rotateY(pi);
+  if (sipm_type_ == "sensl")
```

Comment:
I would put this rotation above, where the other properties of the SiPM are set, so we can use the same `if` condition.